tests/std/tests/P0896R4_counted_iterator/test.cpp

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <cassert>
+#include <concepts>
+#include <iterator>
+
+#include <range_algorithm_support.hpp>
+using namespace std;
+
+struct instantiator {
+    template <input_or_output_iterator Iter>
+    static constexpr void call() {
+        if constexpr (is_permissive) { // TRANSITION, FIXME (File an issue to investigate this later?)
+            (void) input_iterator<Iter>;
+        }
+
+        int input[5] = {1, 2, 3, 4, 5};
+        // [counted.iter.const]
+        {
+            [[maybe_unused]] counted_iterator<Iter> defaultConstructed{};
+
+            [[maybe_unused]] counted_iterator<Iter> constructed(Iter{input}, iter_difference_t<Iter>{2});
+            [[maybe_unused]] counted_iterator<Iter> constructedEmpty{Iter{input}, iter_difference_t<Iter>{0}};
+
+            //[[maybe_unused]] counted_iterator<Iter> convertibleConstructed{};
+            //[[maybe_unused]] counted_iterator<Iter> convertibleAssigned{};
+        }
+
+        // [counted.iter.access]
+        {
+            if constexpr (copy_constructible<Iter>) {
+                const counted_iterator<Iter> lvalueBase{Iter{input}, 2};
+                const same_as<Iter> auto base1 = lvalueBase.base();
+                assert(base1.peek() == Iter{input}.peek());

```

Comment:
`Iter{input}.peek()` is a complicated spelling of `input`. (Several occurrences.)

Reply:
🤦 

That said it kind of fits into the recursion scheme doesnt it?

Reply:
Fixed

Reply:
I suppose we could change all occurrences to `std::addressof(*Iter{input}.peek())`? :trollface: